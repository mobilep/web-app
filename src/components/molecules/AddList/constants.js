export const keyMap = {
	ENTER: 13,
	UP_ARROW: 38,
	DOWN_ARROW: 40,
	TAB: 9,
};

export default {
	keyMap,
};
