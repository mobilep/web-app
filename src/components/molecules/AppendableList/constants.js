export const LIST_TYPES = {
	CHECK: 'check',
	STEP: 'step',
};

export default {
	LIST_TYPES,
};
