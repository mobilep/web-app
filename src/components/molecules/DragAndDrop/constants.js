export const DRAG_START_EVENTS = ['dragover', 'dragenter', 'dragstart'];
export const DRAG_END_EVENTS = ['dragleave', 'dragend', 'drop'];
