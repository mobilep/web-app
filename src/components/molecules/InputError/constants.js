export const POSITIONS = {
	TOP_LEFT_POP_UP: 'topLeftPopUp',
	TOP_RIGHT_POP_UP: 'topRightPopUp',
	BOTTOM_RIGHT_POP_UP: 'bottomRightPopUp',
	BOTTOM_LEFT_POP_UP: 'bottomLeftPopUp',
	RIGHT_POP_UP: 'rightPopUp',
	LEFT_POP_UP: 'leftPopUp',
};

export const COLOR = {
	RED_ICON: 'red',
	WHITE_ICON: 'white',
};

export default {
	POSITIONS,
	COLOR,
};
