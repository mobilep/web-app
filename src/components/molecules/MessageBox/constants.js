export const keyMap = {
	ENTER: 'enter',
	BACKSPACE: 'backspace',
};

export default {
	keyMap,
};
