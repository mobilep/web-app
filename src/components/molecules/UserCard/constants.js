export const CARD_TYPES = {
	NEW_MESSAGE_CARD: 'NEW_MESSAGE_CARD',
	CONVERSATION_CARD: 'CONVERSATION_CARD',
};

export default {
	CARD_TYPES,
};
