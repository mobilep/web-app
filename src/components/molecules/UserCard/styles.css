.UserCard {
	height: 71px;
	display: flex;
	align-items: center;
	position: relative;
}

.UserCard-headerLeftContent {
	min-width: 0;
	flex-grow: 1;
	margin-right: 10px;
}

.UserCard-right-content {
	display: flex;
	flex-direction: column;
	flex-shrink: 0;
	justify-content: center;
}

.UserCard:hover,
.UserCard-active {
	cursor: default;
}

.UserCard-content {
	display: flex;
	flex-direction: column;
	width: calc(100% - 59px);
	margin-left: 10px;
}

.UserCard-header {
	display: flex;
	font-size: 14px;
	width: 100%;
	justify-content: space-between;
}

.UserCard-message {
	font-size: 12px;
	width: 80%;
	margin-top: 4px;
	color: var(--grey3);
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
}

.UserCard-message .MultilineEllipsis {
	max-width: 95%;
}

.UserCard-person {
	font-size: 16px;
	color: var(--almostBlack);
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
}

.UserCard-active .UserCard-person {
	font-weight: 700;
}

.UserCard-date {
	color: var(--grey3);
	font-size: 12px;
}

.UserCard-date-unread {
	color: var(--tomatoRed);
}

.UserCard-evaluation-icon {
	width: 15px;
	height: 16px;
	position: absolute;
	color: var(--grey3);
	right: 20px;
	margin-top: 22px;
}

.UserCard-evaluation-mark {
	position: absolute;
	display: inline-block;
	font-size: 12px;
	right: 40px;
	color: var(--grey3);
	margin-top: 22px;
}

.UserCard-practice {
	height: 78px;
}

.UserCard-scenario {
	font-size: 9px;
	color: var(--grey3);
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis;
	width: 80%;
}

.UserCard-practice-header {
	margin-bottom: 4px;
}

@media (max-width: 768px) {
	.UserCard-header {
		font-size: 16px;
	}

	.UserCard-message {
		font-size: 14px;
		width: 70%;
	}

	.UserCard-evaluation-icon {
		margin-top: 28px;
	}

	.UserCard-evaluation-mark {
		margin-top: 28px;
	}
}
