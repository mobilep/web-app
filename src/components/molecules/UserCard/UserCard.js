import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withLanguageContext } from '../../../utils/LanguageContext';
import { isToday, format } from 'date-fns';
import classNames from 'classnames/bind';

import { CARD_TYPES } from './constants';
import { Avatar, MultilineEllipsis, SendAgo } from '../../atoms';
import { EvaluationIcon } from '../../../icons';

import './styles.css';

class UserCard extends Component {

	static CARD_TYPES = CARD_TYPES;

	formatDate (date, languageCode) {
		const formattedDate = isToday(new Date(date))
			? format(new Date(date), 'HH:mm', { locale: languageCode })
			: format(new Date(date), 'd MMM', { locale: languageCode });
		return formattedDate;
	}

	getClassName (practiceCard, active) {
		return classNames('UserCard', 'listItem', {
			'UserCard-practice': practiceCard,
			'UserCard-active': active,
			'listItem-active': active,
		});
	}

	render () {
		const {
			personInitials,
			person,
			message,
			date,
			active,
			unread,
			avatar,
			type,
			avatarColor,
			practiceCard,
			evaluated,
			evaluationMark,
			className,
			contextSettings,
			waitingForFeedback,
		} = this.props;

		const { dateFnsLanguageCode } = contextSettings;
		const isConversationCard = type === CARD_TYPES.CONVERSATION_CARD;

		const { matches } = window.matchMedia('(max-width: 768px)');

		return (
			<div className={`${this.getClassName(practiceCard, active)} ${className && className}`}>
				<div className={practiceCard ? 'UserCard-avatar' : ''}>
					<Avatar
						initials={personInitials}
						border={unread}
						img={avatar}
						color={`#${avatarColor}`}
						size={!matches ? Avatar.AVATAR_SIZES.AVATAR_28 : Avatar.AVATAR_SIZES.AVATAR_44}
					/></div>
				<div className='UserCard-content'>
					<div className={`UserCard-header ${practiceCard ? 'UserCard-practice-header' : ''}`}>
						<div className='UserCard-headerLeftContent'>
							{waitingForFeedback && practiceCard &&
								<SendAgo date={date} />
							}
							<div className='UserCard-person'>{person}</div>
						</div>
						{isConversationCard
							? <div className='UserCard-right-content'>
								<div className={`UserCard-date ${unread ? 'UserCard-date-unread' : ''}`}>
									{this.formatDate(date, dateFnsLanguageCode)}
								</div>
								{practiceCard && evaluated && <div>
									<span className='UserCard-evaluation-mark'>{evaluationMark}</span>
									<div className='UserCard-evaluation-icon'>
										<EvaluationIcon />
									</div></div>}
							</div>
							: null}
					</div>
					{isConversationCard
						? <div className='UserCard-message'>
							<MultilineEllipsis textContent={(message && message.text) || message} />
						</div>
						: null}
				</div>
			</div>
		);
	}
}

UserCard.propTypes = {
	active: PropTypes.bool,
	avatar: PropTypes.string,
	avatarColor: PropTypes.string,
	className: PropTypes.string,
	content: PropTypes.object,
	contextSettings: PropTypes.object,
	date: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	evaluated: PropTypes.bool,
	evaluationMark: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
	lastIncoming: PropTypes.bool,
	message: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
	person: PropTypes.string,
	personInitials: PropTypes.string,
	practiceCard: PropTypes.bool,
	type: PropTypes.oneOf(Object.values(CARD_TYPES)),
	unread: PropTypes.bool,
	waitingForFeedback: PropTypes.bool,
};

export default withLanguageContext(UserCard);
