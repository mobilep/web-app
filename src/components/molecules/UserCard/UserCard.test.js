import React from 'react';
import { shallow } from 'enzyme';
import mockMatchMedia from '../../../helpers/test/matchMedia';

import UserCard from './UserCard';
import { CARD_TYPES } from './constants';

const CARD_TYPES_ARRAY = Object.values(CARD_TYPES);

describe('Molecule - UserCard', () => {

	beforeAll(() => mockMatchMedia());

	CARD_TYPES_ARRAY.forEach((type) => {
		it(`Should render with type ${type}`, () => {
			const wrapper = shallow(<UserCard personInitials='AB' type={type} />);
			expect(wrapper).toMatchSnapshot();
		});
	});

});
