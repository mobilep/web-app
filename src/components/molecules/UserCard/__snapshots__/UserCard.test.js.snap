// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecule - UserCard Should render with type CONVERSATION_CARD 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`Molecule - UserCard Should render with type NEW_MESSAGE_CARD 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
